### PR TITLE
Emit C enum declarations

### PR DESCRIPTION
**Pull request by VelocityRa**

### CODE REVIEW

File: stdump.cpp [hunk 1]
```diff
@@ -130,16 +130,27 @@ void print_types(Program& program, SymbolTable& symbol_table) {
...
 
-					const StabsSymbol stab_sym = parse_stabs_symbol(full_symbol.c_str());
+					if(verbose)
+						printf("*** PARSING %s\n", full_symbol.c_str());
+
+					const StabsSymbol stab_sym = parse_stabs_symbol(full_symbol.c_str(), verbose);
 					const std::string sym_name = stab_sym.name;
 
 					if (declared_symbols.count(sym_name) == 0) {
+						auto aggregate_max_name_size = [](const auto& fields) -> size_t {
```
Comment (VelocityRa) [inline stdump.cpp:148]: accidentally put this in the wrong commit while rebasing, sorry. hopefully not too much of an issue

File: ccc/stabs.cpp [hunk 2]
```diff
@@ -49,25 +49,30 @@ static StabsType parse_type(const char*& input) {
...
 				std::string name = eat_identifier(input);
 				expect_s8(input, ':', "identifier");
 				s64 value = eat_s64_literal(input);
-				type.enum_type.values.emplace_back(name, value);
+				type.enum_type.fields.emplace_back(name, value);
 				verify(eat_s8(input) == ',',
 					"error: Expecting ',' while parsing enum, got '%c' (%02hhx).",
 					*input, *input);
 			}
+			std::sort(type.enum_type.fields.begin(), type.enum_type.fields.end(),
```
Comment (chaoticgd) [inline ccc/stabs.cpp:65]: This needs the algorithm header.

File: stdump.cpp [hunk 3]
```diff
@@ -113,24 +116,77 @@ void print_symbols(Program& program, SymbolTable& symbol_table) {
...
+							break;
+						}
+						}
+					}
+
 					prefix = "";
 				}
 			}
+			if (sym.storage_type == SymbolType::TYPEDEF) {
+				__debugbreak();
```
Comment (chaoticgd) [inline stdump.cpp:188]: This is Microsoft specific.

**Review by chaoticgd — CHANGES_REQUESTED** [review]

File: stdump.cpp (hunk 3 above)
Comment (VelocityRa) [reply stdump.cpp:188]: did not intend to push this, was debugging

**Review by chaoticgd — APPROVED** [review]

**Pull request merged**